pvlib/tools.py

```diff
@@ -469,3 +472,15 @@ def _first_order_centered_difference(f, x0, dx=DX, args=()):
     # removal in scipy 1.12.0
     df = f(x0+dx, *args) - f(x0-dx, *args)
     return df / 2 / dx
+
+
+def dataset(dataset):
+    """
+    Return a filepath to a dataset bundled with PVLIB with name `dataset`.
+    This utility is intended to be used in tests and examples.
+    """
+    dataset = os.path.join(pvlib.__path__[0], 'data', dataset)
```

Comment:
Should we use pathlib for this?  Doesn't really matter, but might as well follow best practice (which I think is pathlib nowadays)

Reply:
I had in mind keeping it as simple as possible, but this [StackOverflow post](https://stackoverflow.com/questions/67112343/pathlib-vs-os-path-join-in-python) suggests using pathlib too. I'll change it.